client/my-sites/controller.js

```diff
@@ -142,26 +141,26 @@ function isPathAllowedForDomainOnlySite( pathname, domainName ) {
 
 function onSelectedSiteAvailable( context ) {
 	const selectedSite = sites.getSelectedSite();
-	const state = context.store.getState();
+	const getState = () => context.store.getState();
```

Comment:
I think that your suggestion invalidates the contract we have with the function `getState`,
Our contract states:
1. `getState` call should be parameterless
2. `getState` call should be done on the redux store, i.e. the `this` should be `context.store`

So the other option is:
```js
const getState = context.store.getState.bind( context.store );
```

I don't mind either way, so if you prefer that later that's cool by me.